iocore/net/quic/QUICFrame.h

```diff
@@ -62,17 +61,19 @@ class QUICFrame
   virtual size_t size() const = 0;
   virtual bool is_probing_frame() const;
   virtual size_t store(uint8_t *buf, size_t *len, size_t limit) const = 0;
-  virtual void reset(const uint8_t *buf, size_t len);
+  virtual void reset(){};
   virtual QUICFrame *split(size_t size);
   virtual int debug_msg(char *msg, size_t msg_len) const;
+  virtual void parse(const uint8_t *buf, size_t len){};
   virtual QUICFrameGenerator *generated_by();
+  bool valid() const;
   LINK(QUICFrame, link);
 
 protected:
   QUICFrame(QUICFrameId id = 0, QUICFrameGenerator *owner = nullptr) : _id(id), _owner(owner) {}
-  const uint8_t *_buf        = nullptr;
-  size_t _len                = 0;
-  const QUICFrameId _id      = 0;
+  size_t _size               = 0;
```

Comment:
The reason I add size here is the the real size, which we decoded from the packet, might be different . For example :  the `0x08, 0x00, 0x00, 0x01`, we decode as `0x01`. But we encode the `0x01` as only 0x01. The size has changed .

Reply:
I understand the point. Another option would be dropping constructors that receive `buf` and returning a read length from `parse()`. But I'm not sure whether we won't need the size later. Let's keep the size for now.